contracts/paratii/VideoStore.sol

```diff
@@ -51,10 +54,35 @@ contract VideoStore is Ownable, Debug {
        paratiiAvatar.transferFrom(buyer, address(paratiiAvatar),  paratiiPart);
        uint256 ownerPart = price.sub(paratiiPart);
        paratiiAvatar.transferFrom(buyer, owner, ownerPart);
+       bytes32 video_hash = sha3(videoId);
+       user_purchases[buyer].push(video_hash);
+       video_sales[video_hash].push(buyer);
        LogBuyVideo(videoId, buyer, price);
        return true;
     }
 
+    function videoPurchased(string videoId, address user) returns(bool) {
+        bytes32 video_hash = sha3(videoId);
+        address[] users = video_sales[video_hash];
+        for (uint i=0;i<users.length;++i) {
+            if (user == users[i]) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function userOwns(address user, string videoId) returns(bool) {
+        bytes32[] video_hashes = user_purchases[user];
+        bytes32 videoId_hash = sha3(videoId);
+        for (uint i=0;i<video_hashes.length;++i) {
```

Comment:
You need to be a aware of the fact that this is a dangerous pattern in solidity. The `video_hashes` can be of arbitrary legnth, and if it is too long, the transaction will run out of gas and the userOwns function will throw an error. 

There are different patterns to work aroudn this. But the first question is: why is `video_hashes` an array and not a mapping? (If it were a mapping, you would not need the `for`-loop here).